web-marketplace/src/components/organisms/ChooseCreditsForm/ChooseCreditsForm.tsx

```diff
@@ -0,0 +1,209 @@
+import { ChangeEvent, MouseEvent, Suspense, useState } from 'react';
+import { SubmitHandler, useWatch } from 'react-hook-form';
+import { CreditsAmount } from 'web-marketplace/src/components/molecules/CreditsAmount/CreditsAmount';
+import Form from 'web-marketplace/src/components/molecules/Form/Form';
+import { useZodForm } from 'web-marketplace/src/components/molecules/Form/hook/useZodForm';
+
+import { TextButton } from 'web-components/src/components/buttons/TextButton';
+import CheckboxLabel from 'web-components/src/components/inputs/new/CheckboxLabel/CheckboxLabel';
+import { Radio } from 'web-components/src/components/inputs/new/Radio/Radio';
+import { RadioGroup } from 'web-components/src/components/inputs/new/RadioGroup/RadioGroup';
+import { Loading } from 'web-components/src/components/loading';
+import Title from 'web-components/src/components/typography/new/Title';
+
+import { PaymentOptions } from './ChooseCreditsForm.PaymentOptions';
+import {
+  ChooseCreditsFormSchema,
+  FormFields,
+} from './ChooseCreditsForm.schema';
+import { PAYMENT_OPTIONS, PaymentOptionsType } from './ChooseCreditsForm.types';
+
+export function ChooseCreditsForm({
```

Comment:
I believe we're missing some props here
For each credit vintage, we should also have the batch denom (eg C73-005-20230101-20250101-001)
We should also provide for each currency the number of credits available